presto-main/src/main/java/com/facebook/presto/server/StatementResource.java

```diff
@@ -118,10 +132,14 @@
 {
     private static final Logger log = Logger.get(StatementResource.class);
 
-    private static final Duration MAX_WAIT_TIME = new Duration(1, SECONDS);
+    private static final Duration DEFAULT_WAIT_TIME = new Duration(1, SECONDS);
+    private static final Duration MAX_WAIT_TIME = new Duration(10, SECONDS);
     private static final Ordering<Comparable<Duration>> WAIT_ORDERING = Ordering.natural().nullsLast();
     private static final long DESIRED_RESULT_BYTES = new DataSize(1, MEGABYTE).toBytes();
 
+    private final ExecutorService dataExecutor = newCachedThreadPool(daemonThreadsNamed("statement-resource-%s"));
```

Comment:
Not constructive comment.. don't we have any reusable pool that would fit here? Maybe work could be handed back to `responseExecutor` or something like that?